### ISSUE TITLE
Completion Behavior Consistency

### ISSUE THREAD

**Issue by chris-b1** [ipython/ipython#9543]
A couple suggestions to make completions more usable and consistent with other systems. I specifically tried Visual Studio, emacs company-mode, and the qtconsole. 

 1. `<Enter>` should only fill the completion, not enter a newline
 2.  Allow `<PageUp>` and `<PageDown>` to make bigger steps through the completions (10 at a time).  qtconsole doesn't do this, but gives you a 2d grid of a completions, which kind of accomplishes the same thing.

**Comment by takluyver** [comment]
Ping @jonathanslenders - we can probably tweak these keybindings on our side, but it's probably useful feedback for PTK as well.

**Comment by Carreau** [comment]
1) is fixed, I'll bump the page up/down to 6.0, maybe earlier.

**Comment by jonathanslenders** [comment]
Yes, I've been thinking about adding these to ptk as well. I wasn't sure about what to consider the good defaults.

**Comment by chris-b1** [comment]
A related thought - it looks like since the beta the completion options can now be in multiple columns.  Like the qtconsole does, it would be nice if left-arrow / right-arrow would moved across these columns.

**Comment by jonathanslenders** [comment]
For left-arrow/right-arrow, see this issue: https://github.com/jonathanslenders/python-prompt-toolkit/issues/147

It is not as trivial as it sounds. It also requires some architectural changes in how prompt-toolkit works internally. (The key bindings and the layout engine are too strongly decoupled right now, and this makes it very hard to create a key binding for which the behaviour depends on the amount of rows/colums that are used to render the completions.) Probably, this will be in one of the following releases of prompt-toolkit, but it's not the highest priority.

**Comment by zpincus** [comment]
It's worth noting that after enabling mouse support (`TerminalInteractiveShell.mouse_support=True`), the scrollwheel of the mouse can be used to page through a multicolumn completion list column by column. Not totally a hands-on-keyboard solution, but better than having to hit down arrow a zillion times to page through a large set of possible completions.

Even so, having some key combination to advance by columns would be nice, but scrollwheel support is a nice stopgap.
